File: Youtube/5 soal/5c.py
import pandas as pd

df1 = pd.read_csv('Customer.csv')
df2 = pd.read_csv('Orders.csv')
# df2 = df2.sort_values('Order ID')
df2['Order Date'] = pd.to_datetime(df2['Order Date'])

df = pd.merge(df1, df2, on='Customer ID', how='inner')
filter = df.groupby('Customer ID')['Order Date'].min().reset_index(name='pembelian pertama')

gabung = pd.merge(df, filter, on='Customer ID', how='inner')
gabung = gabung[gabung['Order Date'] == gabung['pembelian pertama']]
coba = gabung[gabung['Customer ID']== 'AA-10375']
# print(coba[['Customer ID', 'Sales']])

gabung = gabung.sort_values('Sales', ascending=False)
hasil = gabung.groupby('Customer ID')['Sales'].first().reset_index().sort_values('Customer ID', ascending=True).reset_index(drop=True)
print(hasil)

File: Youtube/5 soal/5c.r
library(dplyr)
library(tidyr)

customer <- read.csv("Analisis excel/Youtube/5 soal/Customer.csv")
print(head(customer))

order <- read.csv("Analisis excel/Youtube/5 soal/Orders.csv")
order$Order.Date <- as.Date(order$Order.Date, "%m/%d/%Y")
print(head(order))


# Inner join customer and order data frames
hasil1 <- inner_join(customer, order, by = "Customer.ID", relationship ="many-to-many")

filter <- order %>% group_by(Customer.ID) %>% summarise(first_order = min(Order.Date))
df <- inner_join(hasil1, filter, by = "Customer.ID", relationship ="many-to-many") %>% filter(first_order == Order.Date)
print(head(df))

cat("\n")
df <- df[order(-df$Sales), ]
hasil2 <- df %>%
    group_by(Customer.ID) %>%
    summarise(total = first(Sales)) %>%
    arrange(Customer.ID)
print(head(hasil2))
print(dim(hasil2))